crates/lakekeeper/src/api/management/v1/tasks.rs

```diff
@@ -77,13 +81,170 @@ pub struct Task {
     pub updated_at: Option<chrono::DateTime<chrono::Utc>>,
 }
 
+impl TryFrom<TaskInfo> for TaskInfoAPI {
+    type Error = ErrorModel;
+
+    fn try_from(value: TaskInfo) -> Result<Self, Self::Error> {
+        let status = value
+            .status()
+            .copied()
+            .map(TaskStatus::from)
+            .or(value.outcome().copied().map(TaskStatus::from))
+            .ok_or(ErrorModel::internal(
+                "Task must have either status or outcome",
+                "StatusOrOutcomeMissing",
+                None,
+            ))?;
+        Ok(TaskInfoAPI {
+            task_id: value.task_id(),
+            project_id: value.project_id().clone(),
+            warehouse_id: value.warehouse_id().ok_or(ErrorModel::internal(
+                "WarehouseListTask needs WarehouseId",
+                "WarehouseIdMissing",
+                None,
+            ))?,
+            queue_name: value.queue_name().clone(),
+            entity: match value.entity_id() {
+                EntityId::Table(table_id) => TaskEntity::Table {
+                    table_id: *table_id,
+                },
+                EntityId::View(view_id) => TaskEntity::View { view_id: *view_id },
+                _ => {
+                    return Err(ErrorModel::internal(
+                        "Only Table and View entities are supported in WarehouseListTask",
+                        "UnsupportedEntityType",
+                        None,
+                    ));
+                }
+            },
+            entity_name: value.entity_name().cloned().ok_or(ErrorModel::internal(
```

Comment:
nit: We have a few unnecessary clones here. This method takes ownership of the original struct. This allows us to just move the ownership of the original object, which is more efficient than creating duplicates and right after dropping the original version.